pkg/logentry/stages/json.go

```diff
@@ -101,35 +101,45 @@ func (j *jsonStage) Process(labels model.LabelSet, extracted map[string]interfac
 
 	if j.cfg.Source != nil {
 		if _, ok := extracted[*j.cfg.Source]; !ok {
-			level.Debug(j.logger).Log("msg", "source does not exist in the set of extracted values", "source", *j.cfg.Source)
+			if Debug {
+				level.Debug(j.logger).Log("msg", "source does not exist in the set of extracted values", "source", *j.cfg.Source)
+			}
```

Comment:
If we think we're going to keep this for any long amount of time, I'd personally prefer wrapping the logger for debugs so we don't have to have a bunch of extra if statements. 

Something like this maybe? 

```
type wrappedDebugLogger struct { l log.Logger }
func (l * wrappedDebugLogger) Log(keyvalues ...string) {
   if Debug { 
      level.Debug(l.l).Log(keyvalues...)
   }
}

func debugLogger(l log.Logger) *wrappedDebugLogger {
   return &wrappedDebugLogger{l: l}
}

... 
debugLogger(j.logger).Log("msg", "source does not exist in the set of extracted values", "source", *j.cfg.Source)
...
```

Reply:
It doesn't look like this would save the allocations:

```
Total:    417.57MB   930.16MB (flat, cum) 64.49%
     75            .          .            
     76            .          .           // Process implements Stage allowing a pipeline stage to also be an entire pipeline 
     77            .          .           func (p *Pipeline) Process(labels model.LabelSet, extracted map[string]interface{}, ts *time.Time, entry *string) { 
     78            .          .           	start := time.Now() 
     79            .          .           	for i, stage := range p.stages { 
     80     320.05MB   320.05MB           		debugLogger(p.logger).Log("msg", "processing pipeline", "stage", i, "name", stage.Name(), "labels", labels, "time", ts, "entry", entry) 
     81            .   512.59MB           		stage.Process(labels, extracted, ts, entry) 
     82            .          .           	} 
     83            .          .           	dur := time.Since(start).Seconds() 
     84      95.51MB    95.51MB           	debugLogger(p.logger).Log("msg", "finished processing log line", "labels", labels, "time", ts, "entry", entry, "duration_s", dur) 
     85            .          .           	if p.jobName != nil { 
     86          2MB        2MB           		p.plDuration.WithLabelValues(*p.jobName).Observe(dur) 
     87            .          .           	} 
     88            .          .           } 
     89            .          .            
     90            .          .           // Name implements Stage 
     91            .          .           func (p *Pipeline) Name() string { 
```

Reply:
ah, that's unfortunate. I think the allocations are from the boxing into `interface{}`.